src/NexusMods.App.UI/Pages/MyGames/MyGamesViewModel.cs

```diff
@@ -41,114 +45,105 @@ public MyGamesViewModel(
-                        vm.AddGameCommand = ReactiveCommand.CreateFromTask(async () =>
-                        {
-                            vm.State = GameWidgetState.AddingGame;
-                            await Task.Run(async () => await ManageGame(loadout.InstallationInstance));
-                            vm.State = GameWidgetState.ManagedGame;
-                        });
-                        vm.RemoveAllLoadoutsCommand = ReactiveCommand.CreateFromTask(async () => 
-                        {
-                            vm.State = GameWidgetState.RemovingGame;
-                            await Task.Run(async () => await RemoveAllLoadouts(loadout.InstallationInstance));
-                            vm.State = GameWidgetState.ManagedGame;
-                        });
+        {
+            // Managed games widgets
+            Loadout.ObserveAll(conn)
+                .Filter(l => l.IsVisible())
+                .RemoveKey()
+                .GroupOn(loadout => loadout.InstallationInstance.LocationsRegister[LocationId.Game])
+                .Transform(group=> group.List.Items.First())
+                .OnUI()
+                .Transform(loadout =>
+                {
+                    var vm = provider.GetRequiredService<IGameWidgetViewModel>();
+                    vm.Installation = loadout.InstallationInstance;
 
-                        vm.ViewGameCommand = ReactiveCommand.Create(
-                            () => { NavigateToLoadout(conn, loadout); }
-                        );
+                    vm.RemoveAllLoadoutsCommand = ReactiveCommand.CreateFromTask(async () =>
+                    {
+                        if (GetJobRunningForGameInstallation(loadout.InstallationInstance).IsT2) return;
 
+                        vm.State = GameWidgetState.RemovingGame;
+                        await Task.Run(async () => await RemoveAllLoadouts(loadout.InstallationInstance));
                         vm.State = GameWidgetState.ManagedGame;
-                        return vm;
-                    })
-                    .Bind(out _managedGames)
-                    .SubscribeWithErrorLogging()
-                    .DisposeWith(d);
-
-                // For the games that are detected, we only want to show those that are not managed, we'll bind directly
-                // to the collection here so we don't need any temporary collections or observables
-                gameRegistry.InstalledGames
-                    .ToObservableChangeSet()
-                    .Except(_managedGames.ToObservableChangeSet().Transform(s => s.Installation))
-                    .OnUI()
-                    .Transform(install =>
+                    });
+
+                    vm.ViewGameCommand = ReactiveCommand.Create(() => { NavigateToLoadout(conn, loadout); });
+
+                    vm.State = ToState(GetJobRunningForGameInstallation(loadout.InstallationInstance), GameWidgetState.ManagedGame);
```

Comment:
This should ignore `AddingGame`, if it is here (managed games) it means the transaction adding the game was already committed, and there is not going to be anything removing the state.